### PR TITLE
refine wavetool.py and check-smart-amplifier test case

### PR DESCRIPTION
**Pull request by aiChaoSONG**
- make calculation of THD+N a generic function
- remove reference in DSM wave analysis.
- discard -r option when call wavetool.py in check-smart-amplifier test case.

### CODE REVIEW

File: tools/wavetool.py [hunk 1]
```diff
@@ -222,6 +205,27 @@ def analyze_wav_wov(wave, fs):
         sys.exit(1002)
     print('wave analysis result: PASSED')
 
+def calc_thdn(wave, fs, freq):
+    """
+    This function calculates THD+N value of a wave.
```
Comment (marc-hb) [inline tools/wavetool.py:210]: Please spell out THD+N at least once. Here seems like a good place.

**Review by marc-hb — COMMENTED** [review]
Good style LGTM, need someone to look at the maths. @singalsu ?

File: tools/wavetool.py (hunk 1 above)
Comment (aiChaoSONG) [reply tools/wavetool.py]: fixed

**Comment by aiChaoSONG** [comment]
@keyonjie @xiulipan There is no new algorithm introduced, maybe you can help to approve?

**Comment by aiChaoSONG** [comment]
@keyonjie Please help to review and approve this PR, I have tested it on TGL-NOCODEC(DSM case) and TGL-HDA(WOV case)

**Review by keyonjie — APPROVED** [review]
lgtm, thanks.

**Review by cujomalainey — APPROVED** [review]

**Pull request merged**